src/Application.php

```diff
@@ -218,12 +163,6 @@ public function finish($callback, $priority = 0)
                 2 => $this,
             ]);
 
-            if ($ret instanceof Response) {
-                $event->setResponse($ret);
-            } elseif (null !== $ret) {
-                throw new \RuntimeException('An after middleware returned an invalid response value. Must return null or an instance of Response.');
-            }
-
```

Comment:
But this is the `after()` method (`RESPONSE` event), you pointed to `finish()` (which is `TERMINATE`).